content/general/functions.textile

```diff
@@ -8,17 +8,176 @@ Reactor Functions allow your server-less functions to be invoked following chann
+
+
+h2. Envelopes
+
+Messages sent by Reactor Functions can be contained within an envelope. This can be useful for clients which are unable to access response headers or status codes.
 
-Reactor Functions are are designed to trigger server-less functions on common platform such as Amazon and Google.  If you are looking to trigger execution of code on your own servers over HTTP when a message is published or event occurs, see our "Reactor WebHooks":/general/webhooks.
 
 h2. Payload types
 
-Ably currently supports three types of data that can be delivered via our Firehose:
+Ably currently supports two types of data that can be delivered via our Functions:
+
+* "Messages":#messages - messages trigger function calls as soon as they are published on a channel
+* "Presence events":#presence - when clients enter, update their data, or leave channels, the presence event triggers a function call
+
+You can "configure Functions":#configure from the Reactor tab in your "account dashboard":https://support.ably.io/solution/articles/3000048664-how-do-i-access-my-account-dashboard on a per app basis which can apply to one or more channels in that app. Reactor Functions can be filtered by channel naming using a regular expression.
+
+
+
+h2(#messages). Messages
+
+If you are unfamiliar with how channels work in Ably, we recommend you first review the "realtime channel documentation":/realtime/channels-messages#channels.
+
+When a message is published on a channel, a Reactor Function message event is triggered, which in turn will send a message to your server-less function.
+
+Every request will contain your message within the @body@ part of the request. The contents of the @body@ depends on the format of your messages, and whether you've used an envelope or not.
+
+* No Envelope - Without an envelope, the contents of @body@ will simply be the contents message that was sent on the Ably Channel. 
+
+* Envelope - If you use an envelope, the @body@ will contain a bunch of extra attributes, which are the following:
+
+
+- source := the source of the request, namely "@channel.message"
+- appId := the ID of your application with Ably
+- channel := name of the channel the Ably Message was published on
+- site := an internal site identifier indicating the data center from which the message was published
+- ruleId := an ID for the rule which has created this request
+- messages := an @Array@ of messages with fields for each message described below
+
+Each member of @messages@ has the following fields:
+
+- id := unique ID assigned by Ably to this message
+- name := a string representing the event name for the published message, see the "publish method":/rest/
+- connectionId := the public unique identifier for the publisher's connection. Find out more about "@connectionId@":/realtime/connection#id
+- timestamp := the time in milliseconds since the epoch when this message was received by Ably
```

Comment:
Question: what is epoch? If it's an event, perhaps replace with `the time in milliseconds between the epoch and the message being received by Ably`?

Reply:
That's a good question, and we do refer to the epoch in many places.  @tomczoink I think it's confusing indeed and something we're familiar with. We should have a suppport article (unless one exists) explaining what the milliseconds since epoch time represents and how to convert that to a time, which ofrtunately in Javascript is just `new Date(timeInMsSinceEpoch)`